project.yaml

```diff
@@ -907,14 +907,47 @@ Layer:
               ('railway_' || (CASE WHEN railway IN ('platform') THEN railway ELSE NULL END)),
               (('aeroway_' || CASE WHEN aeroway IN ('runway', 'taxiway', 'helipad') THEN aeroway ELSE NULL END))
             ) AS feature
-          FROM planet_osm_polygon
-          WHERE highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'living_street', 'track', 'path', 'platform', 'services')
-            OR railway IN ('platform')
-            OR aeroway IN ('runway', 'taxiway', 'helipad')
+          FROM (
+          SELECT 
+              way, highway, railway, aeroway, z_order, way_area FROM planet_osm_polygon
+            WHERE highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'living_street', 'track', 'path', 'platform', 'services')
+              OR railway IN ('platform')
+              OR aeroway IN ('taxiway', 'helipad')
+          UNION ALL
+          SELECT -- runways with width either tagged or estimated from runway length
+              ST_Buffer(
```

Comment:
I hadn't realized that this was all in a ST_Buffer.

I would **normally** suggest 

```sql
ST_Transform(ST_Buffer(ST_Transform(way,4326)::geography,
  CASE
    WHEN width ~ '^\d{1,3}(\.\d+)?$' THEN 0.5*width::real
    ELSE LEAST(GREATEST(ST_Length(st_transform(way,4326)::geography)/50,12.0),75.0)
  END,
  'endcap=flat join=round')::geometry,ST_SRID(way)
```

But geography st_buffer doesn't take `buffer_style_parameters`, so we need something else, and it has to work on older versions of PostGIS. Fortunately, we can do this, because `st_buffer(geography, double precision)` is `geography(ST_Transform(ST_Buffer(ST_Transform(geometry($1), _ST_BestSRID($1)), $2), 4326))`

```sql
SELECT -- aeroways, converted to areas
    way,
    feature,
    ST_Area(way) AS way_area
  FROM
    (SELECT 
        ST_Transform(
          ST_Buffer(
            ST_Transform( -- Use a projection using real meters
              way,
              _ST_BestSRID(ST_Transform(way, 4326)::geography) -- Find the SRID to use
            ),
            0.5 * CASE -- Buffer by the half-width
              WHEN width ~ '^\d{1,3}(\.\d+)?$' -- Validate the width tag
                THEN LEAST(width::real, 150) -- Place a maximum on the width
              WHEN aeroway = 'runway' -- No valid width, so estimate from length if a runway
                LEAST(GREATEST(ST_Length(st_transform(way,4326)::geography)/50,12.0),75.0)
              ELSE --taxiway
                6
            END,
            'endcap=flat join=round'
          ),
          ST_SRID(way)
        ) AS way,
        'aeroway_' || aeroway AS feature
      WHERE aeroway IN ('runway', 'taxiway')
        AND way && !bbox!) AS p -- doesn't quite work, for reasons below
```

A mistake in what you have is that it prevents index usage, so `way && !bbox!` is needed, but once the `!bbox!` token is used, it needs to be used everywhere in the query because it's no longer automatically added by Mapnik.

Aeroways are also more complicated because we're doing a st_buffer that may buffer up to 150 meters, or whatever the width is being capped at. We can't rely on the normal buffering of the metatile extents, because it's only 128px, and at high zooms, 150m is well over that.

Instead of `way && !bbox!` for aeroways, we need
```sql
way && ST_Transform(
  ST_Expand(
    ST_Transform(
      !bbox!,
      _ST_BestSRID(ST_Transform(!bbox!,4326)::geography)),
    150),
  ST_SRID(!bbox!))
```